packages/store/addon/-private/system/references/record.ts

```diff
@@ -170,11 +170,9 @@ export default class RecordReference extends Reference {
      @return {Promise<record>} the record for this RecordReference
   */
   reload() {
-    let record = this.value();
-    if (record) {
-      return record.reload();
+    if (this._id !== null) {
+      return this.store.findRecord(this.type, this._id, { reload: true });
```

Comment:
since you cant reload a record you've created client side this works well (unlike value above). Once #7104 lands though we should find by identifier.